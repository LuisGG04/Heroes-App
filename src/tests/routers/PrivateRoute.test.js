import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/authContext";
import { PrivateRoute } from "../../routers/PrivateRoute";

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    Navigate: () => <span>Sin autenticar</span>
}));

describe("Pruebas en la ruta PrivateRoute", () => {

    Storage.prototype.setItem = jest.fn();

    test("Debe de mostrar el componente si estÃ¡ autenticado y guardar en localStorage", () => {
        
        const contextValue = {
            user: {
                logged: true,
                name: "Luis"
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={["/"]}>
                    <PrivateRoute>
                        <h1>Private Component</h1>
                    </PrivateRoute>
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.text().trim()).toBe("Private Component");
        expect(localStorage.setItem).toHaveBeenCalledWith("lastPath", "/");
    });

    test("Debe que debe de estar autenticado", () => {
        
        const contextValue = {
            user: {
                logged: false
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={["/"]}>
                    <PrivateRoute>
                        <h1>Private Component</h1>
                    </PrivateRoute>
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper.find("span").exists()).toBe(true);
    });
    
});
