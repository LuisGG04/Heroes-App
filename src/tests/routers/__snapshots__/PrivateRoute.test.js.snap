// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en la ruta PrivateRoute Debe de mostrar el componente si est√° autenticado y guardar en localStorage 1`] = `
<h1>
  Private Component
</h1>
`;
